agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/GeneRepository.java

```diff
@@ -76,19 +57,42 @@ public Gene getOneGene(String primaryKey) {
 		HashMap<String, String> map = new HashMap<>();
 
 		map.put("primaryKey", primaryKey);
-		String query = " MATCH p1=(q:Species)-[:FROM_SPECIES]-(g:Gene) WHERE g.primaryKey = $primaryKey "
-			+ "OPTIONAL MATCH p2=(g:Gene)--(:SOTerm) "
-			+ "OPTIONAL MATCH p3=(g:Gene)--(:Synonym) "
-			+ "OPTIONAL MATCH p4=(g:Gene)--(:SecondaryId) "
-			+ "OPTIONAL MATCH p5=(g:Gene)--(:GenomicLocation) "
-			+ "OPTIONAL MATCH p6=(g:Gene)--(:CrossReference) "
-			+ "RETURN p1, p2, p3, p4, p5, p6";
+		String query = """
+			MATCH (q:Species)-[:FROM_SPECIES]-(g:Gene {primaryKey: $primaryKey})
+			OPTIONAL MATCH (g)-[:ANNOTATED_TO]-(soTerm:SOTerm)
+			WITH g, q, COLLECT(soTerm) AS soTerms
+			OPTIONAL MATCH (g)-[:ALSO_KNOWN_AS]-(synonym:Synonym)
+			WITH g, q, soTerms, COLLECT(synonym) AS synonyms
+			OPTIONAL MATCH (g)-[:ALSO_KNOWN_AS]-(secondaryId:SecondaryId)
+			WITH g, q, soTerms, synonyms, COLLECT(secondaryId) AS secondaryIds
+			OPTIONAL MATCH (g)-[:ASSOCIATION]-(genomicLocation:GenomicLocation)
+			WITH g, q, soTerms, synonyms, secondaryIds, COLLECT(genomicLocation) AS genomicLocations
+			OPTIONAL MATCH (g)-[:CROSS_REFERENCE]->(crossRef:CrossReference)
+			RETURN q AS Species,
+			       soTerms AS SOTerms,
+			       synonyms AS Synonyms,
+			       secondaryIds AS SecondaryIds,
+			       genomicLocations AS GenomicLocations,
+			       g AS Gene,
+			       COLLECT(crossRef) AS CrossReferences
+			""";
 
 		Iterable<Gene> genes = query(query, map);
-		for (Gene g : genes) {
-			if (g.getPrimaryKey().equals(primaryKey)) {
-				return g;
-			}
+		if (genes.iterator().hasNext()) {
+			Gene gene = genes.iterator().next();
```

Comment:
@cmpich you need to do the `if (g.getPrimaryKey().equals(primaryKey)) {` check sometimes for whatever reason two genes match and this iterator will be looking at the wrong one.